src/services/storage/handlers/offer.ts

```diff
@@ -35,42 +75,17 @@ const handler: Handler<StorageServices> = {
       logger.info(`Created new StorageOffer for ${provider}`)
     }
 
-    switch (event.event) {
-      case 'TotalCapacitySet':
-        offer.totalCapacity = event.returnValues.capacity
-        logger.info(`Updating capacity ${offer.totalCapacity} (ID: ${offer.address})`)
-        break
-      case 'MessageEmitted': {
-        const msg = event.returnValues.message
-
-        if (!msg || msg.length === 0) {
-          break
-        }
-
-        const [firstMsg, ...restMsg] = msg
-        const flag = firstMsg.substring(2, 4)
-
-        if (flag === '01') { // PeerId definition
-          offer.peerId = decodeByteArray([`0x${firstMsg.substring(4)}`, ...restMsg])
-          logger.info(`PeerId ${offer.peerId} defined (ID: ${offer.address})`)
-        } else {
-          throw new EventError(`Unknown message flag ${flag}!`, event.event)
-        }
-        break
-      }
-      case 'BillingPlanSet':
-        await updatePrices(offer, event.returnValues.period, event.returnValues.price)
-        break
-      default:
-        logger.error(`Unknown event ${event.event}`)
-        break
+    if (!handlers[event.event]) {
+      logger.error(`Unknown event ${event.event}`)
     }
 
+    await handlers[event.event](event, offer, offerService)
+
     await offer.save()
 
-    // @TODO if we receive an unknown message then update actually doesn't happens, because offer don't changed
-    // Should we emit `update` event then???
-    if (offerService.emit) offerService.emit(created ? 'created' : 'updated', offer.toJSON())
+    if (offerService.emit && created) {
+      offerService.emit('created', offer.toJSON())
```

Comment:
Well with this refactor, I believe we will get both `created` and `updated` events for the first event of the Offer right? Maybe we can just simply ditch the `created` for now and see if    the need for it raises later on?
I will leave that up to you to decide :-)

Reply:
hmm... you are right.